src/server/controllers/users.controller.js

```diff
@@ -163,32 +165,49 @@ export function localSignIn(req, res, next) {
       }
     })
     .catch(function (err) {
-      res.status(400).json(err.message);
+      res.status(400).json({ err_msg: err.message });
     });
 }
 
 export function requestSecretCode(req, res, next) {
-  let date = new Date();
-  let dateString = date.toISOString();
-  let cipher = crypto.createCipher('aes192', req.body.email);
-  let secretCode = cipher.update(dateString, 'utf-8', 'hex');
-  secretCode += cipher.final('hex');
-
-  User.findOne({ email: req.body.email }).exec()
-    .then(user => {
-      if (!user) {
-        throw new Error(userCallback.ERR_USER_NOT_FOUND);
-      } else {
-        // TODO: Save secret code to db and check validaion of it. Only the last one is valid.
-        mailingController.sendEmail(req.body.email, mailStrings.RESETPW_SUBJECT,
-          mailStrings.RESETPW_HTML, secretCode);
+  if (req.body.email) {
+    User.findOne({ email: req.body.email }).exec()
+      .then(user => {
+        if (!user) {
+          throw new Error(userCallback.ERR_USER_NOT_FOUND);
+        } else {
+          SecretCode.findOne({ email: req.body.email, isValid: true }).exec()
+            .then(validSecretCode => {
```

Comment:
If you return `SecretCode.findOne({ email: req.body.email, isValid: true }).exec()` here, 
you can use non-nested promise!